db_migration/db_diff.py

```diff
@@ -27,151 +29,249 @@ def __init__(self, connection, table, columns):
         self.columns = columns
         self.results = []
 
-    def fetch(self, condition, args, joins=()):
-        cols = ', '.join('{}.{}'.format(self.table, col) for col in self.columns)
-        j = ' '.join('JOIN {tbl} AS {as} ON {as}.{col} = {val}'.format(**join) for join in joins)
-        q = 'SELECT DISTINCT {} FROM {} {} WHERE {} ORDER BY {}.{} LIMIT ALL'
+    def fetch_all(self):
+        with self.connection.cursor() as cursor:
+            q = "SELECT {} FROM {}".format(", ".join(self.columns), self.table)
+            # print("Executing query:", q)
+            cursor.execute(q)
+            self.results = cursor.fetchall()
+
+    def fetch(
+        self, condition, args, joins=(), clear_results=True, order_by="ASC", limit="ALL"
+    ):
+        if clear_results:
+            self.results = []
+
+        cols = ", ".join("{}.{}".format(self.table, col) for col in self.columns)
+        j = " ".join(
+            "JOIN {tbl} AS {as} ON {as}.{col} = {val}".format(**join) for join in joins
+        )
+        q = "SELECT DISTINCT {} FROM {} {} WHERE {} ORDER BY {}.{} {} LIMIT {}".format(
+            cols, self.table, j, condition, self.table, self.columns[0], order_by, limit
+        )
+
+        # Print the query template with placeholders
+        print("Executing query template:", q)
+        print("With arguments:", args)
+
+        # Replace placeholders with actual argument values for debugging
+        query_with_args = q
+        for arg in args:
+            if isinstance(arg, str):
+                arg = "'{}'".format(
+                    arg.replace("'", "''")
+                )  # Escape single quotes in strings
+            elif arg is None:
+                arg = "NULL"
+            query_with_args = query_with_args.replace("%s", str(arg), 1)
```

Comment:
I think it should be `psycopg2`'s job to convert and quote values is SQL properly…